index.js

```diff
@@ -40,8 +41,35 @@ export async function run() {
     const { context } = github
     const octokit = github.getOctokit(githubToken)
 
-    const pullRequestNumber = context.payload.pull_request.number
-    const pullRequestBody = context.payload.pull_request.body
+    if (!context.payload.pull_request && !prNumber) {
+      throw new Error(
+        'You must either trigger this action from a pull request, or manually set the `prNumber` input\n' +
+          'Please check your setup: https://github.com/ivangabriele/find-and-replace-pull-request-body#usage',
+      )
+    }
+
+    if (context.payload.pull_request && prNumber) {
+      throw new Error(
+        "You can't use `prNumber` while in the context of a pull request event.\n" +
+          'Please check your setup: https://github.com/ivangabriele/find-and-replace-pull-request-body#usage',
+      )
+    }
+
+    let pullRequestNumber
```

Comment:
I'm not a big fan of mutable variables, I prefer the code to be as functional and immutable as possible (using `const`) but it's not a deal-breaker for this PR ^^.

Reply:
I also prefer a functional style, although in this case I’d say it’s more of a conditional assignment rather than mutation, as it’s never mutated after being set and both values are primitives 

If we want to avoid making the API call on L65 unless necessary I think it means either:
- keeping it how it is
- Moving the logic into a function like `getPullRequestDetails` that returns early to avoid mutation
- encapsulating the logic that follows it in a function, then calling it from each of the 2 code branches


Im happy to do a bit of refactoring and break it all up into functions, but I didn’t want to step on any toes and tried to make the PR diff as minimal as possible

Reply:
@mcky No you did a nice job, that's definitely cleaner! You can update your PR here. I will update the README on my side.

Reply:
@ivangabriele I won’t be able to merge the refactor until Monday (as I want to double check it actually works in production) so I can raise that as a follow-up PR to not block this one if you prefer?

or I can merge it with the caveat it hasn’t been properly tested (manually or programmatically) if you’d want to handle that - I don’t want to potentially break things for the sake of style though 😅

Reply:
I understand! What I'll do then is merging this PR right now and add a few more tests before releasing ^^